query_execution/QueryContext.hpp

```diff
@@ -302,8 +307,8 @@ class QueryContext {
    *
    * @return True if valid, otherwise false.
    **/
-  bool isValidJoinHashTableId(const join_hash_table_id id) const {
-    return id < join_hash_tables_.size();
+  bool isValidJoinHashTableId(const join_hash_table_group_id gid, const join_hash_table_id id = 0) const {
```

Comment:
Please don't use a default value here. Also, please update the `doxygen` comments for updated arguments.